### PR TITLE
Prevent setting LSF_SERVER to an undefined environment variable

### LINKED ISSUE #6031: Detect if empty environment variable is used for LSF_SERVER
In earlier versions, it was common to replace LSF_SERVER using an environment variable in the configuration.
This is still being used by some users, but we have to make sure that the environment variable is not empty.

`QUEUE_OPTION LSF_SERVER $MY_LSF_SERVER`

That specific issue is described more detailed here:
Relates to https://github.com/equinor/ert/issues/5715

### PR DESCRIPTION
**Pull request by larsevj**
**Issue**
Resolves #6031 

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured new behaviour is tested (opened GUI? screenshots? tried workflows?)

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/config/queue_config.py [hunk 1]
```diff
@@ -90,7 +90,14 @@ def from_dict(cls, config_dict: ConfigDict) -> QueueConfig:
                     f"Valid choices are {sorted(VALID_QUEUE_OPTIONS[queue_system])}."
                 )
             if values:
-                queue_options[queue_system].append((option_name, values[0]))
+                if option_name == "LSF_SERVER" and values[0].startswith("$"):
+                    raise ConfigValidationError(
+                        f"Invalid server name for QUEUE_OPTION LSF LSF_SERVER: "
+                        f"{values[0]}. Check for empty enviroment variable. "
```
Comment (oysteoh) [inline src/ert/config/queue_config.py:96]: enviroment -> environment

"The LSF_SERVER is usually specified in the site-config file. "  - maybe more like something like this? 
"The LSF_SERVER value is usually provided by default in a site-configuration file. If you have no special needs you can safely remove it from your configuration"

File: src/ert/config/queue_config.py [hunk 2]
```diff
@@ -91,6 +91,14 @@ def from_dict(cls, config_dict: ConfigDict) -> QueueConfig:
                     f"Valid choices are {sorted(VALID_QUEUE_OPTIONS[queue_system])}."
                 )
             if values:
+                if option_name == "LSF_SERVER" and values[0].startswith("$"):
+                    raise ConfigValidationError(
+                        "Invalid server name for QUEUE_OPTION LSF LSF_SERVER:"
+                        f" {values[0]}. Check for empty environment variable. The"
```
Comment (andreas-el) [inline src/ert/config/queue_config.py:97]: Is the environment variable empty? Or is it just not set?

File: src/ert/config/queue_config.py [hunk 3]
```diff
@@ -91,6 +91,14 @@ def from_dict(cls, config_dict: ConfigDict) -> QueueConfig:
                     f"Valid choices are {sorted(VALID_QUEUE_OPTIONS[queue_system])}."
                 )
             if values:
+                if option_name == "LSF_SERVER" and values[0].startswith("$"):
+                    raise ConfigValidationError(
+                        "Invalid server name for QUEUE_OPTION LSF LSF_SERVER:"
+                        f" {values[0]}. Check for an undefined environment variable. "
+                        "The LSF_SERVER value is usually provided by default in a"
+                        " site-configuration file. If you have no special needs you"
+                        " can safely remove it from your configuration."
```
Comment (andreas-el) [inline src/ert/config/queue_config.py:100]: This is obviously up for discussion but you could consider something like this?
```
"Invalid server name specified for QUEUE_OPTION LSF LSF_SERVER:"
f" {values[0]}. Server name is currently an undefined environment variable. "
"The LSF_SERVER keyword is usually provided by the site-configuration file, "
"beware that you are effectively replacing the default value provided."
```

**Review by andreas-el — CHANGES_REQUESTED** [review]
Please take a look at the feedback provided.

PR-name and commit is done perfectly. 🚀

**Review by andreas-el — APPROVED** [review]
You nailed it! 🚀

**Pull request merged**